packages/openapi3/src/schema-emitter-3-0.ts

```diff
@@ -206,12 +209,39 @@ export class OpenAPI3SchemaEmitter extends OpenAPI3SchemaEmitterBase<OpenAPI3Sch
           (schema instanceof Placeholder || "$ref" in schema) &&
           !(type && shouldInline(program, type))
         ) {
-          if (type && (type.kind === "Model" || type.kind === "Scalar")) {
+          if (type && type.kind === "Model") {
             return new ObjectBuilder({
               type: "object",
               allOf: Builders.array([schema]),
               ...additionalProps,
             });
+          } else if (type && type.kind === "Scalar") {
+            let objectInitializer: Record<string, unknown> = {};
+            const stdScalar = $.scalar.getStdBase(type);
+            if (stdScalar?.name) {
+              // already has same std scalar member, use $ref
+              if (stdScalarNames.has(stdScalar.name) && "$ref" in schema) {
```

Comment:
can you not use identity instead of name matching, what if someone make a new scalar with the same name

Reply:
Yes it can https://typespec.io/playground/?c=c2NhbGFyIHN0cmluZzs%3D&e=%40typespec%2Fopenapi3&options=%7B%7D

You have to get the scalar reference in the compiler and check they match or at least if you use string check the parent namespace is TypeSpec